src/internal/obj/uniform_client.go

```diff
@@ -0,0 +1,58 @@
+package obj
+
+import (
+	"context"
+	"io"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+var _ Client = &uniformClient{}
+
+// uniformClient is for ensuring uniform behavior across all the object clients
+type uniformClient struct {
+	c Client
+}
+
+func newUniformClient(c Client) Client {
+	if c == nil {
```

Comment:
I kind of think this shouldn't be a thing. We could error / panic if it is nil, but I think that would add a lot of bloat to our code if we followed that pattern. I think we should instead just use the pattern you suggested in the datum ID PR where we only create a wrapper client when the client being wrapped is not nil. That would allow us to get rid of these extra checks that just add bloat.